src/main/java/com/teamabnormals/savage_and_ravage/core/other/SREvents.java

```diff
@@ -142,14 +142,14 @@ public static void onRightClickEntity(PlayerInteractEvent.EntityInteract event)
 			Player player = event.getEntity();
 			CompoundTag persistentData = target.getPersistentData();
 			if (!persistentData.getBoolean(POISON_TAG)) {
-				if (target.level.random.nextDouble() < SRConfig.COMMON.poisonPotatoChance.get()) {
```

Comment:
Shouldn't this be using the `RandomSource` instance of the `target` local variable?

Reply:
Technically no as the player interact event runs always on server thread so there's no worry of multithreaded issues. This is a case where server level random is safe. Basically if the code you're in has a chance of running off thread, do not use the level's random. Most people forget the entity constructor and finalizeSpawn method in entities can be ran during worldgen which is offthread and is why those have to use the entity's randomsource and not the level's

Reply:
Ok, I was only checking because I assumed you were only changing stuff necessary/relevant to the fix.
Don't include any changes to the `SREvents` class, then.